internal/querynode/query_collection.go

```diff
@@ -543,176 +570,66 @@ func (q *queryCollection) receiveQueryMsg(msg queryMsg) error {
 			zap.Any("serviceTime", st),
 			zap.Any("delta seconds", (guaranteeTs-serviceTime)/(1000*1000*1000)),
 			zap.Any("msgID", msg.ID()),
-			zap.String("msgType", msgTypeStr),
+			zap.Any("msgType", msg.Type()),
 		)
 		msg.RecordSpan()
-		q.addToUnsolvedMsg(msg)
-		sp.LogFields(
-			oplog.String("send to unsolved buffer", "send to unsolved buffer"),
-			oplog.Object("guarantee ts", gt),
-			oplog.Object("serviceTime", st),
-			oplog.Float64("delta seconds", float64(guaranteeTs-serviceTime)/(1000.0*1000.0*1000.0)),
-		)
-		sp.Finish()
-		return nil
-	}
-	tr.Record("get searchable time done")
-
-	log.Debug("doing query in receiveQueryMsg...",
-		zap.Int64("collectionID", collectionID),
-		zap.Any("sm.GuaranteeTimestamp", gt),
-		zap.Any("serviceTime", st),
-		zap.Any("delta seconds", (guaranteeTs-serviceTime)/(1000*1000*1000)),
-		zap.Int64("msgID", msg.ID()),
-		zap.String("msgType", msgTypeStr),
-	)
-	switch msgType {
-	case commonpb.MsgType_Retrieve:
-		err = q.retrieve(msg)
-	case commonpb.MsgType_Search:
-		err = q.search(msg)
-	default:
-		err = fmt.Errorf("receive invalid msgType = %d", msgType)
-		return err
+		return false, nil
 	}
-	tr.Record("operation done")
+	return true, nil
+}
 
-	if err != nil {
-		publishErr := q.publishFailedQueryResult(msg, err.Error())
-		if publishErr != nil {
-			finalErr := fmt.Errorf("first err = %s, second err = %s", err, publishErr)
-			return finalErr
-		}
-		log.Debug("do query failed in receiveQueryMsg, publish failed query result",
-			zap.Int64("collectionID", collectionID),
-			zap.Int64("msgID", msg.ID()),
-			zap.String("msgType", msgTypeStr),
-		)
-		return err
-	}
-	log.Debug("do query done in receiveQueryMsg",
-		zap.Int64("collectionID", collectionID),
-		zap.Int64("msgID", msg.ID()),
-		zap.String("msgType", msgTypeStr),
-	)
-	tr.Elapse("all done")
-	sp.Finish()
-	return nil
+// TODO @xiaocai2333: implement it.
+func (q *queryCollection) mergeSearchReq(mergedMsg queryMsg, msg queryMsg) queryMsg {
+	return msg
 }
 
 func (q *queryCollection) doUnsolvedQueryMsg() {
 	log.Debug("starting doUnsolvedMsg...", zap.Any("collectionID", q.collectionID))
+	emptySearchQueue := true
+	go func() {
```

Comment:
what 's  the purpose of this goroutine?

Reply:
When there is no task to perform for the goroutine for searching `doSearchRequests`, I hope that a search request can be sent quickly. The purpose of this goroutine is to make the goroutine ` doUnsolvedQueryMsg` know  that the goroutine `doSearchRequests` wants a search requests.